server/src/dtos/system-config.dto.ts

```diff
@@ -269,9 +270,12 @@ class SystemConfigMachineLearningDto {
   @ValidateBoolean()
   enabled!: boolean;
 
-  @IsUrl({ require_tld: false, allow_underscores: true })
+  @IsUrl({ require_tld: false, allow_underscores: true }, { each: true })
+  @ArrayMinSize(1)
+  @Transform(({ value }) => (Array.isArray(value) ? value : [value]))
   @ValidateIf((dto) => dto.enabled)
-  url!: string;
+  @ApiProperty({ type: 'array', items: { type: 'string', format: 'uri' }, minItems: 1 })
```

Comment:
I think you don't actually need to specify api property. Have you tried removing it?

Reply:
It's just to make the OpenAPI property more detailed. `format` and `minItems` aren't in the spec without this.

Reply:
Oh that's interesting, okay. I would've hoped it'd pick those up from class validator